src/SPC/builder/linux/LinuxBuilder.php

```diff
@@ -203,18 +209,22 @@ public function buildPHP(int $build_target = BUILD_TARGET_NONE): void
 
         $this->cleanMake();
 
-        if (($build_target & BUILD_TARGET_CLI) === BUILD_TARGET_CLI) {
+        if ($enableCli) {
             logger()->info('building cli');
             $this->buildCli($extra_libs, $use_lld);
         }
-        if (($build_target & BUILD_TARGET_FPM) === BUILD_TARGET_FPM) {
+        if ($enableFpm) {
             logger()->info('building fpm');
             $this->buildFpm($extra_libs, $use_lld);
         }
-        if (($build_target & BUILD_TARGET_MICRO) === BUILD_TARGET_MICRO) {
+        if ($enableMicro) {
             logger()->info('building micro');
             $this->buildMicro($extra_libs, $use_lld, $cflags);
         }
+        if ($enableEmbed) {
+            logger()->info('building embed');
+            $this->buildEmbed($extra_libs, $use_lld);
+        }
```

Comment:
It is better to compile `buildMicro` as the last one. Because phpmicro sapi may have some php source code patches that may affect other sapis.

Reply:
I thought about that, but I fear those patches may cause issues when using libphp directly as FrankenPHP does. Some (most) of them doesn't look necessary for these use case. With the current approach, you have both of the best worlds: "vanilla" PHP by default, and the Micro patches as long as you add `--build-micro`.

Reply:
I reviewed my code again, and I found no matter whether micro is built or not, some patches will be performed, such as static opcache support, sapi name check of internal extensions (sqlite, readline, zend_accelerator, proc). 

These patches are currently common to all SAPIs, so they will be executed when phpmicro is decompressed. And of course, there are also some non-universal patches, such as Phar that allows micro to support compression, micro's support in Windows environment, etc. I have placed these in `$builder->buildMicro()` to handle them. 

In short, there should be no problems here now. 👍